pkg/cluster/mpi/statefulset.go

```diff
@@ -118,14 +118,29 @@ func (c statefulSetComponent) Reconcile(ctx *core.Context) (ctrl.Result, error)
 
 func (c statefulSetComponent) Finalize(ctx *core.Context) (ctrl.Result, bool, error) {
 	cr := objToMPICluster(ctx.Object)
+
 	pvcListOpts := []client.ListOption{
 		client.InNamespace(cr.Namespace),
 		client.MatchingLabels(meta.MatchLabels(cr)),
 	}
-
 	err := actions.DeleteStorage(ctx, pvcListOpts)
+	if err != nil {
+		return ctrl.Result{}, false, fmt.Errorf("cannot delete storage: %w", err)
+	}
+
+	sts := &appsv1.StatefulSet{
```

Comment:
I guess this was getting deleted due to the owner reference before?

Reply:
This data structure is just a "stub" used to point to a service that will be deleted.

Reply:
Yes, I'm referring to the fact that you are now explicitly deleting the StatefulSet. 

Also: Isn't this the purpose of ObjectKeys? Do you know why we are using these partially populated resources to indicate what should be deleted?

Reply:
I need to check this out... In my understanding, the `deleteIfExists` function expects a `StatefulSet` as a "pointer" to what is supposed to be deleted.

Reply:
The k8s client does need an `Object` here, not `ObjectKeys`: 
sigs.k8s.io/controller-runtime@v0.10.0/pkg/client/interfaces.go:67